packages/block-library/src/post-author/edit.js

```diff
@@ -1,31 +1,253 @@
+							onChange={ ( size ) => {
+								props.setAttributes( {
+									avatarSize: Number( size ),
+								} );
+							} }
+						/>
+					) }
+					<ToggleControl
+						label={ __( 'Show bio' ) }
+						checked={ showBio }
+						onChange={ () =>
+							props.setAttributes( { showBio: ! showBio } )
+						}
+					/>
+				</PanelBody>
+				<PanelBody title={ __( 'Text settings' ) }>
+					<FontSizePicker
+						value={ fontSize.size }
+						onChange={ setFontSize }
+					/>
+				</PanelBody>
+			</InspectorControls>
+
+			{ InspectorControlsColorPanel }
+
+			<BlockControls>
+				<AlignmentToolbar
+					value={ align }
+					onChange={ ( nextAlign ) => {
+						props.setAttributes( { align: nextAlign } );
+					} }
+				/>
+				<BlockColorsStyleSelector
+					TextColor={ TextColor }
+					BackgroundColor={ BackgroundColor }
+				>
+					{ ColorPanel }
+				</BlockColorsStyleSelector>
+			</BlockControls>
+
+			<TextColor>
+				<BackgroundColor>
+					<div
+						ref={ ref }
+						className={ classnames( blockClassNames, {
+							[ `has-text-align-${ align }` ]: align,
+						} ) }
+						style={ blockInlineStyles }
+					>
+						{ showAvatar && authorDetails && (
+							<div className="wp-block-post-author__avatar">
+								<img
+									width={ avatarSize }
+									src={
+										authorDetails.avatar_urls[ avatarSize ]
+									}
+									alt={ authorDetails.name }
+								/>
+							</div>
+						) }
+						<div className="wp-block-post-author__content">
+							{ ( ! RichText.isEmpty( byline ) ||
+								isSelected ) && (
+								<RichText
+									className="wp-block-post-author__byline"
+									multiline={ false }
+									placeholder={ __( 'Write byline …' ) }
+									withoutInteractiveFormatting
+									allowedFormats={ [
+										'core/bold',
+										'core/italic',
+										'core/strikethrough',
+									] }
+									value={ byline }
+									onChange={ ( value ) =>
+										props.setAttributes( { byline: value } )
+									}
+								/>
+							) }
+							<p className="wp-block-post-author__name">
+								{ authorDetails.name }
+							</p>
+							{ showBio && (
+								<p className={ 'wp-block-post-author__bio' }>
+									{ authorDetails.description }
+								</p>
+							) }
+						</div>
+					</div>
+				</BackgroundColor>
+			</TextColor>
+		</>
 	);
-	return author ? (
-		<address>
-			{ sprintf(
-				/* translators: %s: author name. */
-				__( 'By %s' ),
-				author.name
-			) }
-		</address>
-	) : null;
 }
 
-export default function PostAuthorEdit() {
-	if ( ! useEntityId( 'postType', 'post' ) ) {
+function PostAuthorEdit( props ) {
+	const [ postAuthor, setPostAuthor ] = useEntityProp(
+		'postType',
+		'post',
+		'author'
+	);
+
+	const [ authorDetails, setAuthorDetails ] = useState( false );
+
+	if ( ! postAuthor ) {
 		return 'Post Author Placeholder';
 	}
-	return <PostAuthorDisplay />;
+
+	if ( ! authorDetails ) {
+		getAuthorDetails( postAuthor, setAuthorDetails );
```

Comment:
Render functions shouldn't have side effects inline like this (triggering a network request).

>“side effects” [...] can’t be done during rendering.

https://reactjs.org/docs/hooks-overview.html#effect-hook

As documented there, it's a good use-case for a hook like `useEffect`.

Reply:
Further, [Core Data module](https://github.com/WordPress/gutenberg/tree/master/packages/core-data) entities are intended to serve this purpose of fetching common data. There's already a selector/resolver which can be used for this.

```js
const authorDetails = useSelect(
	( select ) => select( 'core' ).getUser( postAuthor ),
	[ postAuthor ]
);
```

Reply:
Both comments https://github.com/WordPress/gutenberg/pull/19894#discussion_r425195717 and https://github.com/WordPress/gutenberg/pull/19894#discussion_r425201413 have been incorporated into the proposed changes at #22359.